com.avaloq.tools.ddk.xtext.builder/src/com/avaloq/tools/ddk/xtext/builder/MonitoredClusteringBuilderState.java

```diff
@@ -632,7 +632,9 @@ public boolean isCanceled() {
           if (resource instanceof XtextResource) {
             ((XtextResource) resource).getCache().clear(resource);
           }
-          storeBinaryResource(resource, buildData);
+          if (resource != null) {

```

Comment:
I do not think this if is needed, because the first thing storeBinaryResource will do is to check resource instanceof ..., and that is only true if resource is not null

Reply:
I am not a fan of this particular change either.
However, previous verify jobs (SpotBugs) complain about it because it is initialized (line 523) as null and it is possible for addResource (line 536) to return null as well.
Do we have an alternative (like suppress checking for this particular case)?

Reply:
Uploaded another change for this one. Thank you.